### PR TITLE
*: use retry to make tests stable.

### PR DESCRIPTION
**Pull request by disksing**

### CODE REVIEW

File: pkg/testutil/testutil.go [hunk 1]
```diff
@@ -58,3 +59,24 @@ func tryAllocTestURL() string {
...
+	waitMaxRetry   = 100
+	waitRetrySleep = time.Millisecond * 50
+)
+
+// CheckFunc is a condition checker that passed to WaitUtil. Its implementation
+// may call c.Fatal() to abort the test, or c.Log() to add more information.
+type CheckFunc func(c *check.C) bool
+
+// WaitUtil repeatly evaluates f() for a period of time, util it returns true.
+func WaitUtil(c *check.C, f CheckFunc) {
```
Comment (siddontang) [inline pkg/testutil/testutil.go]: do we need to supply another `WaitUtilWIth(c *check.C, f CheckFunc, retryCount int, waitSleep Duration)` to let outer use customized retry count and sleep time?

Comment (disksing) [inline pkg/testutil/testutil.go]: Not need for now. Not difficult to add it later.

File: server/cluster_worker_test.go [hunk 2]
```diff
@@ -467,23 +465,30 @@ func (s *testClusterWorkerSuite) TestHeartbeatSplit2(c *C) {
...
+			if res == nil {
+				c.Log("no response")
+				return false
+			}
+			if res.GetChangePeer() == nil || res.GetChangePeer().GetChangeType() != pdpb.ConfChangeType_AddNode {
+				c.Log("response is not AddNode")
+				return false
+			}
+			s.onChangePeerRes(c, res.GetChangePeer(), r1)
+			return true
```
Comment (nolouch) [inline server/cluster_worker_test.go]: `return len(r1.Peers) == len(s.stores)` in here to reduce nesting?

Comment (disksing) [inline server/cluster_worker_test.go]: The inner loop adds one peer each iteration, we need to add 4 peers here, so I guess add a nested level will be more clear.

Comment (nolouch) [inline server/cluster_worker_test.go]: got it

File: server/cluster_worker_test.go [hunk 3]
```diff
@@ -467,23 +465,30 @@ func (s *testClusterWorkerSuite) TestHeartbeatSplit2(c *C) {
...
-	for {
-		resp := s.heartbeatRegion(c, s.clusterID, r1, leaderPeer, false)
-		if resp == nil {
-			break
-		}
-		s.checkChangePeerRes(c, resp.GetChangePeer(), pdpb.ConfChangeType_AddNode, r1)
-	}
+	testutil.WaitUntil(c, func(c *C) bool {
+		testutil.WaitUntil(c, func(c *C) bool {
+			res := s.heartbeatRegion(c, s.clusterID, r1, leaderPeer)
```
Comment (huachaohuang) [inline server/cluster_worker_test.go]: Seems there is a pattern here, maybe you can extract a function like:
```
func checkHeartbeatRegion(region, leader, AddNode/RemoveNode) bool
```

**Comment by huachaohuang** [comment]
LGTM

**Pull request merged**